fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HiveMetaStoreCache.java

```diff
@@ -561,6 +562,93 @@ public void invalidatePartitionCache(ExternalTable dorisTable, String partitionN
         }
     }
 
+    /**
+     * Selectively refreshes cache for affected partitions based on update information from BE.
+     * This method optimizes cache refresh by only invalidating specific partitions
+     * rather than the entire table cache.
+     *
+     * @param table The Hive table whose partitions were modified
+     * @param partitionUpdates List of partition updates from BE
+     */
+    public void refreshAffectedPartitions(HMSExternalTable table,
+            List<org.apache.doris.thrift.THivePartitionUpdate> partitionUpdates) {
+        if (partitionUpdates == null || partitionUpdates.isEmpty()) {
+            return;
+        }
+
+        NameMapping nameMapping = table.getOrBuildNameMapping();
+        long tableId = Util.genIdByName(nameMapping.getLocalDbName(), nameMapping.getLocalTblName());
+
+        // Classify partition updates by type
+        Set<List<String>> modifiedPartitions = new HashSet<>();
+        boolean hasNewPartitions = false;
+
+        for (org.apache.doris.thrift.THivePartitionUpdate update : partitionUpdates) {
+            List<String> partitionValues = HiveUtil.toPartitionValues(update.getName());
+
+            switch (update.getUpdateMode()) {
+                case APPEND:
+                case OVERWRITE:
+                    modifiedPartitions.add(partitionValues);
+                    break;
+                case NEW:
+                    hasNewPartitions = true;
+                    break;
+                default:
+                    LOG.warn("Unknown update mode {} for partition {}",
+                            update.getUpdateMode(), update.getName());
+                    break;
+            }
+        }
+
+        // Refresh file cache for modified partitions
+        if (!modifiedPartitions.isEmpty()) {
+            invalidateFileCache(nameMapping, tableId, modifiedPartitions);
+        }
+
+        // Refresh partition values cache if new partitions were created
+        if (hasNewPartitions) {
+            invalidatePartitionValuesCache(nameMapping);
```

Comment:
Can we just modify the cache value instead of invalidate it all?
Otherwise, for insert with new partition, it always invalidate all cache